File: packages/csharp/src/V2/StatusClient.cs
using System.Threading;
using System.Threading.Tasks;
using Tesote.Sdk.Internal;
using Tesote.Sdk.Models;

namespace Tesote.Sdk.V2;

/// <summary>v2 status + whoami endpoints.</summary>
public sealed class StatusClient
{
    private readonly Transport _transport;

    internal StatusClient(Transport transport)
    {
        _transport = transport;
    }

    /// <summary>Public status check.</summary>
    public async Task<StatusResponse> GetAsync(CancellationToken ct = default)
    {
        var node = await _transport.RequestAsync(RequestOptions.Get("/v2/status"), ct).ConfigureAwait(false);
        return Json.Deserialize<StatusResponse>(node);
    }

    /// <summary>Identifies the bearer-token holder.</summary>
    public async Task<WhoamiResponse> WhoamiAsync(CancellationToken ct = default)
    {
        var node = await _transport.RequestAsync(RequestOptions.Get("/v2/whoami"), ct).ConfigureAwait(false);
        return Json.Deserialize<WhoamiResponse>(node);
    }
}

File: packages/java/src/main/java/com/tesote/sdk/v2/StatusClient.java
package com.tesote.sdk.v2;

import com.fasterxml.jackson.databind.JsonNode;
import com.tesote.sdk.Transport;
import com.tesote.sdk.internal.Json;
import com.tesote.sdk.models.Status;
import com.tesote.sdk.models.Whoami;

/**
 * v2 status + whoami. Mirrors v1; separate path prefix.
 */
public final class StatusClient {
    private final Transport transport;

    public StatusClient(Transport transport) { this.transport = transport; }

    public Status status() {
        JsonNode node = transport.request(Transport.Options.get("/v2/status"));
        return Json.treeToValue(node, Status.class);
    }

    public Whoami whoami() {
        JsonNode node = transport.request(Transport.Options.get("/v2/whoami"));
        return Json.treeToValue(node, Whoami.class);
    }
}
